docs(docs-infra): adds note according to Symlink problem

FrederikSchlemmer opened the pull request:
Closes #24709

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #24709 

Executing "yarn setup" could fail under Windows because of Symlinks problems. There isn't any hint for a solution currently.


## What is the new behavior?
Executing "yarn setup" could also fail under Windows because of Symlinks problems. 

There is a hint for the solution to execute the needed windows scripts, which will fix the Symlink problem and "yarn setup" executes without errors.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

gkalpak commented on aio/README.md:
```diff
@@ -105,6 +105,8 @@ The general setup is as follows:
 
 * Open a terminal, ensure the dependencies are installed; run an initial doc generation; then start the doc-viewer:
 
+>If linking problems occur under Windows, please execute the scripts located at `angular/scripts/windows` and retry. (admin rights required)
```
This here is the "Partial doc generation for editors" section. The issue is not specific to partial doc generation; it's a general issue 😒

To give this more visibility, can you please create a section right above ` ## Using ServiceWorker locally`. Something like:

```markdown
## Developing on Windows

The `packages/` directory may contain Linux-specific symlinks, that are not recognized by Windows. This can cause the docs generation process to fails, because it cannot locate certain files.

To fix this, you need to first run `scripts/windows/create-symlinks.sh`. This will create some temporary files, where the symlinks used to be. Make sure to not commit those files along with your changes.

When you are done, you can restore the original symlinks by running `scripts/windows/remove-symlinks.sh`.
```

jenniferfell replied:
Agree re Developing on Windows as a clear section. 

Suggested edits: 

The `packages/` directory may contain Linux-specific symlinks, which are not recognized by Windows.
These unresolved links cause the docs generation process to fail because it cannot locate certain files.
To fix this problem, run `scripts/windows/create-symlinks.sh`. This command creates temporary files where the symlinks used to be. Make sure not to commit those files with your documentation changes. When you are done making and testing your documentation changes, you can restore the original symlinks by running `scripts/windows/remove-symlinks.sh`.

jenniferfell replied:
I don't fully understand why I would restore the original symlinks. What problems are caused by running create-symlinks.sh and resolved by remove-symlinks.sh?

gkalpak replied:
The "problem" is the temporary files created by `create-sumlinks.sh`. They show up as local changes in your git working copy, which means you cannot do certain operations (such as rebasing etc) that are only allowed when your git working copy is clean.

FrederikSchlemmer replied:
Thanks for the review and the feedback! I created a section "Developing on Windows" and picked the suggested content.
To avoid any possible confusion about the necessary removing process of the temporary files I added a short explanation.

gkalpak replied:
This line is unnecessarily duplicating info now. If you absolutely feel like mentioning something here, just add a link to the "Developing on Windows" section.

gkalpak reviewed (changes requested):
BTW, this is one of the rare cases, where the commit message should actually start with `docs(docs-infra): ...`. Don't miss out on this rare opportunity :smiley:

jenniferfell reviewed (commented):
Thanks for making Windows users happier.

gkalpak commented on aio/README.md:
```diff
@@ -43,6 +43,15 @@ Here are the most important tasks you might need to use:
+The `packages/` directory may contain Linux-specific symlinks, which are not recognized by Windows.
+These unresolved links cause the docs generation process to fail because it cannot locate certain files.
+
+To fix this problem, run `scripts/windows/create-symlinks.sh`. This command creates temporary files where the symlinks used to be. Make sure not to commit those files with your documentation changes.
```
Can you also mention that one needs admin rights (or [Dev Mode enabled][1])?

[1]: https://blogs.windows.com/buildingapps/2016/12/02/symlinks-windows-10/#1ydZJ9eThBkAykfP.97

FrederikSchlemmer replied:
Added a hint! Maybe [Windows Developer Mode](https://docs.microsoft.com/en-us/windows/uwp/get-started/enable-your-device-for-development)  should be a link?

gkalpak replied:
+1 on making it a link.

gkalpak reviewed (changes requested):
Thx for making the changes, @FrederikSchlemmer 👍 
A couple more to go :smiley:

gkalpak reviewed (approved):
Othan making `Windows Developer Mode` a link, it LGTM (as long as it LGT @jenniferfell).

FrederikSchlemmer committed:
docs(docs-infra): adds note according to Symlink problem

docs: adds note according to Symlink problem

Closes #24709
docs(docs-infra): adds section "Developing on Windows"


Merge remote-tracking branch 'origin/aioREADME' into aioREADME


docs(docs-infra): adds information about admin rights


docs(docs-infra): adds hint


docs(docs-infra): Change to link

FrederikSchlemmer commented:
@jenniferfell Do you got any final thoughts about this PR?

gkalpak commented:
@jenniferfell, is this ready to go?
